scalableminds/webknossos
# Implement folder specific search for datasets

- folder-specific search with optional recursion
- folder path displayed as tag

![image](https://user-images.githubusercontent.com/2486553/206155057-392b6f2e-727f-466b-bf1f-2cb42e8144e1.png)
![image](https://user-images.githubusercontent.com/2486553/205608125-34e25363-31b6-4d98-8f1b-b9c9b4959387.png)

### URL of deployed dev instance (used for testing):
- https://folderspecificsearch.webknossos.xyz

### Steps to test:
- create some folders to  have a folder hierarchy
- move some datasets
- use the global search to find a dataset
- customize the search to only  search a specific folder and check that the search-subfolders option works
- check that the paths next to the dataset names look good
- switching to another folder while the search is active should change the folder that is searched (earlier versions cleared the search instead)
- if the global search is enabled, there shouldn't be an active folder. activating a folder in that case, should switch automatically to the folder-specific search.
- when going from global search to folder-specific search, the most recently active folder should be activated again (the same goes for clearing the search box)

### Issues:
- contributes to #6669

------
(Please delete unneeded items, merge only when none are left open)
